### PR TITLE
fix(detail): Deprecate URL construction for detail page.

### PR DESCRIPTION
**Pull request by vikram-raj**
<strong>Note: If there are pending changes to the PR, prefix the PR title with "WIP" and add the label "DO NOT MERGE"</strong>

### Mandatory
 - [ ] What does this PR do? 
Remove URL cunstruction function

- [ ] What issue/task does this PR references?
https://github.com/openshiftio/openshift.io/issues/3845

- [ ] Are the tests Included?
NA

### CODE REVIEW

**Comment by alien-ike** [comment]
### Ike Plugins (test-keeper)

Thank you @vikram-raj for this contribution!



It appears that no tests have been added or updated in this PR.

Automated tests give us confidence in shipping reliable software. Please add some as part of this change.

If you are an admin or the reviewer of this PR and you are sure that no test is needed then you can use the command `/ok-without-tests` as a comment to make the status green.


Your plugin configuration is stored in the [file](https://github.com/fabric8-ui/fabric8-planner/blob/e80accc3d59799335a7afdcb595f9dd1256d6ad2/.ike-prow/test-keeper.yaml).

File: src/app/effects/work-item.effects.ts [hunk 1]
```diff
@@ -112,7 +112,7 @@ export class WorkItemEffects {
                 return new WorkItemActions.GetChildren(parent);
               } else {
                 if (payload.openDetailPage) {
-                  this.router.navigateByUrl(this.router.url.split('plan')[0] + 'plan/detail/' + wItem.number,
+                  this.router.navigateByUrl(document.location.pathname + '/detail/' + wItem.number,
```
Comment (sanbornsen) [inline src/app/effects/work-item.effects.ts]: Is it the best solution? Can we avoid `document`?

**Comment by jarifibrahim** [comment]
The commit message is `fiX...`, it should be `fix...`

**Comment by vikram-raj** [comment]
> The commit message is fiX..., it should be fix...

@jarifibrahim Done.

**Pull request merged**